scrapysplashwrapper/crawler.py

```diff
@@ -69,6 +69,12 @@ def __init__(self, url: str, useragent: str, cookies: Optional[List[Dict[Any, An
                 cookies = []
             self.cookies: List[Dict[Any, Any]] = cookies
             self.referer: str = referer
+            if proxy.strip():
+                parsed_proxy = urlparse(proxy)
+                self.proxy_hostname = parsed_proxy.hostname
+                self.proxy_port = parsed_proxy.port
+                self.proxy_scheme = parsed_proxy.scheme
```

Comment:
we may want to add some validation: in that case, I'm not sure what will happen if  `parsed_proxy.port` is  `None` (the lua script might fail). It should never happen, but that's the kind of thing that will be super frustrating for a user if they get a crash instead of a warning.

Reply:
Yes, I need to check that the proxy field is not empty and that the different attributes are valid as well, and print warnings in case they are not.